### DISCUSSION TITLE
[FEATURE REQUEST] add SecureJsonData argument in grafana4_datasource state and module

### DISCUSSION THREAD

**Discussion by keenobi** [saltstack/salt#67305] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

We want to create a datasource with an HTTP header. But in Grafana's API call the header value must be provided in the `secureJsonData` field. https://grafana.com/docs/grafana/latest/developers/http_api/data_source/#create-a-data-source

**Describe the solution you'd like**

We would like to have a secure_json_data attribute to be able to set the desired values.

**Additional context**

We will propose the PR soon.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-3av8jjyf-oBQ2M0vhXOhJpNpRkPWBvg)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!
